data_model/src/permission.rs

```diff
@@ -91,10 +97,10 @@ impl PermissionTokenSchema {
 
 impl PermissionToken {
     /// Construct [`Self`]
-    pub fn new<T: Encode>(definition_id: PermissionTokenId, payload: &T) -> Self {
+    pub fn new(definition_id: PermissionTokenId, payload: serde_json::Value) -> Self {
         Self {
             definition_id,
-            payload: payload.encode(),
+            payload: StringWithJson(payload),
```

Comment:
Why not to convert`Value` to the `String` inside constructor? Because common pattern for this struct is `.to_string()` before actual use which might be costly.

Reply:
The main point is `Serialize` and `Deserialize` derives. I'd like it to be consistent with other json in `genesis.json`. It will be weird if it will be string in json format inside another json

Reply:
I'll change the name

Reply:
> The main point is `Serialize` and `Deserialize` derives. I'd like it to be consistent with other json in `genesis.json`. It will be weird if it will be string in json format inside another json

We can provide our own `Serialize` and `Deserialize` which will do conversion `Serialized` -> `serde_json::Value` -> `String` (and wise versa). What do you think?

Reply:
And what's the point of doing that if it would be the same, we just move conversion from `Encode/Decode` to `Serialize/Deserialize`?

While having `serde_json::Value` protects us from currupting json with out actions, which `String` cannot provide

Reply:
> And what's the point of doing that if it would be the same, we just move conversion from Encode/Decode to Serialize/Deserialize?

We can validate that string which we received is valid json without actually parsing it:

```rust
let _: serde::de::IgnoredAny = serde_json::from_str(data)?;
```

> While having serde_json::Value protects us from currupting json with our actions, which String cannot provide

Which actions may corrupt permission token payload?

And we anyway have to deserialize/decode permission token inside validator, so we can verify that we received valid json.

Reply:
Check out new code